### ISSUE TITLE
Missing OIDs for common DN objects

### ISSUE THREAD

**Issue by bkreider** [smallstep/cli#204]
### Subject of the issue

For a swiss army knife of X509 certificates, it's odd inspect certificate code leaves out the OIDs shown in the [RFC 2253](https://tools.ietf.org/html/rfc2253) example of common fields:
```
                    CN      commonName
                    L       localityName
                    ST      stateOrProvinceName
                    O       organizationName
                    OU      organizationalUnitName
                    C       countryName
                    STREET  streetAddress
                    DC      domainComponent
                    UID     userid
```

Here is what the code supports: https://github.com/smallstep/cli/blob/master/pkg/x509/pkix/pkix.go
```
	"2.5.4.6":  "C",
	"2.5.4.10": "O",
	"2.5.4.11": "OU",
	"2.5.4.3":  "CN",
	"2.5.4.5":  "SERIALNUMBER",
	"2.5.4.7":  "L",
	"2.5.4.8":  "ST",
	"2.5.4.9":  "STREET",
	"2.5.4.17": "POSTALCODE",
```

### Steps to reproduce
I have a private client cert I was inspecting, but you can view the source code to see which x.500 entities are supported.

### Expected behaviour
I would expect that DC and UID objects would be supported before STREET.  

### Actual behaviour
My Subject looks like this:
```
Subject: UnknownOID=0.9.2342.19200300.100.1.25,UnknownOID=0.9.2342.19200300.100.1.25,UnknownOID=0.9.2342.19200300.100.1.25,UnknownOID=0.9.2342.19200300.100.1.25,OU=Users,UnknownOID=0.9.2342.19200300.100.1.44
```

Using Openssl it looks like this:
```
openssl x509  -noout -subject -in C16B0CE23D5C29CE.pem
subject= /DC=com/DC=<redacted>/DC=ad/DC=<redacted>/OU=Users/0.9.2342.19200300.100.1.44=user1
```
Note: Openssl doesn't use the UID short string either. 

### Additional notes

I am using client certs for mutual TLS, but also using the subject for authentication.

**Comment by dopey** [comment]
Hey @bkreider thanks for opening an issue! 

First off, the tagline for the repo is aspirational ;p But, more seriously we've been dogfood-ing our own certs (which don't use those OIDs) so long that we just didn't notice they were missing. 

This shouldn't be difficult to add, but it may be a little while until we're able to deliver it as part of an official release. 

Again, thanks for pointing this out. Surprised this is the first time it's been brought up :)

Also, the x509 package you linked above has been removed from our codebase in the most recent branch we've been working on (called `ssh`). It was a copy of the official crypto `x509` package but with ed25519 support added. Now that golang x509 supports ed25519 we were able to remove our dupe of the package.
